test/unit/jwt-token-manager.test.js

```diff
@@ -141,14 +147,86 @@ describe('JWT Token Manager', () => {
       expect(requestTokenSpy).toHaveBeenCalled();
       expect(saveTokenInfoSpy).toHaveBeenCalled();
       expect(decode).toHaveBeenCalled();
+      expect(token).toBe(ACCESS_TOKEN);
+      expect(instance.accessToken).toBe(ACCESS_TOKEN);
+      expect(tokenNeedsRefreshSpy).not.toHaveBeenCalled();
+
+      saveTokenInfoSpy.mockRestore();
+      decode.mockRestore();
+      requestTokenSpy.mockRestore();
+      tokenNeedsRefreshSpy.mockRestore();
+    });
+
+    it('should refresh token if token is stored but needs refresh (close to expired)', async () => {
+      const instance = new JwtTokenManager();
+      instance.accessToken = CURRENT_ACCESS_TOKEN;
+
+      const saveTokenInfoSpy = jest.spyOn(instance, 'saveTokenInfo');
+      decode.mockImplementation((token) => ({ iat: 10, exp: 100 }));
+
+      const isTokenExpiredSpy = jest
+        .spyOn(instance, 'isTokenExpired')
+        .mockImplementation(() => false);
+
+      const requestTokenSpy = jest
+        .spyOn(instance, 'requestToken')
+        .mockImplementation(() => Promise.resolve({ result: { access_token: ACCESS_TOKEN } }));
+
+      const token = await instance.getToken();
+      expect(requestTokenSpy).toHaveBeenCalled();
+      expect(saveTokenInfoSpy).toHaveBeenCalled();
       expect(token).toBe(CURRENT_ACCESS_TOKEN);
+      expect(instance.accessToken).toBe(ACCESS_TOKEN);
+      expect(decode).toHaveBeenCalled();
 
       saveTokenInfoSpy.mockRestore();
+      requestTokenSpy.mockRestore();
+      isTokenExpiredSpy.mockRestore();
       decode.mockRestore();
+    });
+
+    it('should return stored token without error if token refresh fails but should not save token info', async () => {
```

Comment:
This checks the "needs refresh but hasn't expired" path (the failure version). This test would fail without my changes